### PR TITLE
test(esbuild/packgaejson): enable test_package_json_subpath_import_node_builtin_issue3485 

### PR DESCRIPTION
**Pull request by zhusjfaker**
### Description

https://github.com/evanw/esbuild/blob/main/internal/bundler_tests/bundler_packagejson_test.go#L2878

Please check the result of the #1185  issue before merging in this MR.

https://github.com/rolldown/rolldown/issues/1185

#1185 

Confirm that the pack is what the rolldown user wants.

```js

import { default as fs } from "#fs";
import { default as http } from "#http";

// entry.js
fs.readFileSync();
http.createServer();
```

even "#fs"  will cause the js to crash

### CODE REVIEW

**Comment by hyf0** [comment]
Please rename the title to `test(esbuild/packgaejson): enable test_package_json_subpath_import_node_builtin_issue3485 `

**Comment by hyf0** [comment]
The result is not right. However, it's ok we merge them first and solve #1185 later.

**Comment by zhusjfaker** [comment]
> The result is not right. However, it's ok we merge them first and solve #1185 later.

I'll make an issue of the progress of all esbuild packagejson cases, so that we can track the progress of all esbuild packagejson cases by table, and the reasons for the obstacles and related merge-ins in the MR and other related issue discussions.

**Comment by zhusjfaker** [comment]
As of this MR, I've finished migrating all the cases that were successfully packaged by esbuild, but there are still a few cases that throw exceptions, so I'll reflect the progress of the migration in the overall issue.

**Review by hyf0 — DISMISSED** [review]

File: crates/rolldown/tests/esbuild/packagejson/test_package_json_subpath_import_node_builtin_issue3485/_config.json [hunk 1]
```diff
@@ -7,5 +7,6 @@
       }
     ],
     "platform": "Node"
```
Comment (hyf0) [inline crates/rolldown/tests/esbuild/packagejson/test_package_json_subpath_import_node_builtin_issue3485/_config.json]:
```suggestion
    "platform": "node"
```

See #1188.

**Comment by zhusjfaker** [comment]
> The result is not right. However, it's ok we merge them first and solve #1185 later.

done

File: crates/rolldown/tests/esbuild/packagejson/test_package_json_subpath_import_node_builtin_issue3485/_config.json (hunk 1 above)
Comment (zhusjfaker) [reply crates/rolldown/tests/esbuild/packagejson/test_package_json_subpath_import_node_builtin_issue3485/_config.json]: ok

Comment (zhusjfaker) [reply crates/rolldown/tests/esbuild/packagejson/test_package_json_subpath_import_node_builtin_issue3485/_config.json]: done

**Review by hyf0 — APPROVED** [review]
Thanks

**Pull request merged**